staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -634,6 +641,19 @@ func (o *DiffOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 		return err
 	}
 
+	if o.Prune {
```

Comment:
I'm guessing here you would only create the pruner if the prune flag is set to true, and then you can use the `nil` condition on the pruner as a way to know if it's enabled or not (and/or do a no-op on a nil pruner).